### PR TITLE
UCT/IB: Set minor gcc version if using clang version >= 300.

### PR DESCRIPTION
**Pull request by alinask**
Also in Jenkins, run ucx_info after compiling with armclang.

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7215/ for details (Mellanox internal link).

File: src/uct/ib/mlx5/ib_mlx5.h [hunk 1]
```diff
@@ -16,6 +16,14 @@
 #include <ucs/debug/log.h>
 #include <ucs/type/status.h>
 
+#ifdef __clang__
+#  define CLANG_VERSION ( __clang_major__ * 100 + __clang_minor__)
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.h]: pls add some doc to why we are doing this

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7220/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7221/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/7224/ for details (Mellanox internal link).

**Comment by alinask** [comment]
@yosefe is this good to go?

**Pull request merged**